libs/wingsdk/src/shared-aws/permissions.ts

```diff
@@ -235,3 +235,36 @@ export function calculateBucketPermissions(
 
   return policies;
 }
+
+export function calculateSecretPermissions(
+  arn: string,
+  target: AwsTarget,
+  ops: string[]
+): { [key: string]: any }[] {
+  let policy = {};
+  let policies = [];
+  if (
+    ops.includes(cloud.SecretInflightMethods.VALUE) ||
+    ops.includes(cloud.SecretInflightMethods.VALUE_JSON)
+  ) {
+    switch (target) {
+      case AwsTarget.AWSCDK:
+        policy = {
+          effect: "Allow",
+          actions: ["secretsmanager:GetSecretValue"],
+          resources: [arn],
+        };
+        break;
+      case AwsTarget.TF_AWS:
+        policy = {
+          effect: "Allow",
+          action: ["secretsmanager:GetSecretValue"],
```

Comment:
I am wondering if TF can accept `actions` and `resources` (plural) and then we can avoid this silly duplication everywhere (copying @marciocadev)

Reply:
I did this test, unfortunately the object used in CDK is different from TF. If you use the same one that is used in CDK, the policy will look like this when compiled to TF.
`"policy": "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Action\":\"none:null\",\"Resource\":\"*\"}]}",`

With `Resource: *` and `Action: none:null`

If we use the TF object to generate the CDK, a compilation error occurs.
```bash
~/code/wing/apps/wing/bin/wing compile -t awscdk cdk.w
preflight error: Validation failed with the following errors:
  [stack/Default/Default/wing-function/Default/ServiceRole/DefaultPolicy] A PolicyStatement must specify at least one 'action' or 'notAction'.
  [stack/Default/Default/wing-function/Default/ServiceRole/DefaultPolicy] A PolicyStatement used in an identity-based policy must specify at least one resource.
```

Reply:
Trying to solve this in #2048